code/espurna/relay.ino

```diff
@@ -241,6 +241,19 @@ void setSpeed(unsigned char speed) {
     }
 }
 
+unsigned char getLight() {
+    unsigned char light = (_relays[0].target_status ? 1 : 0);
+    return light;
+}
+
+void setLight(unsigned char light) {
+    if ((0 <= light) & (light <= 1)) {
```

Comment:
Isn't this essentially renaming `relay/0` topic to simply `light`? We can use relayParsePayload to parse toggle and some strings too, payload becoming either 0 (off), 1 (on) or 2 (toggle). relayStatusWrap directly accepts it's result.

And unless I am missing something, did you mean && instead of &?

Reply:
Yes. Is specific for the **Sonoff iFan02**: the relay 0 is `light`.
I don't know that we can use `relayParsePayload`.
This function is a _copy and paste_ from `setSpeed` and is just the same at this line: https://github.com/xoseperez/espurna/blob/dev/code/espurna/relay.ino#L234

Reply:
But the first one is real though, I think we totally can.

About speeds, see the speed bitmap above. Each speed is one combination of relays, so we are using AND to set desired speed setting.
We should probably introduce Rotary Switch type for this (i.e. more that 0 or 1 for state)

Reply:
What is your proposal to do now?

Reply:
Second part is suggestion in general, not for this pr.

The way that setSpeed works is not really applicable here. We can just use relayParsePayload when parsing MQTT / API string and pass it to helper function that I mentioned: https://github.com/xoseperez/espurna/blob/0987e0195041b39109f775865ac16bd14b2801a6/code/espurna/relay.ino#L861